### ISSUE TITLE
Creating a new swap chain for immersive mode on Daydream fails

### ISSUE THREAD

**Issue by bluemarvin** [MozillaReality/FirefoxReality#121]
When creating a new swap chain on Daydream devices, rendering to the swap chain will intermittently fail. When the original swap chain is recreated it seems to work fine again.

**Comment by MortimerGoro** [comment]
I was able to reproduce a similar issue in Oculus Go this way:

* Enter the WebVR sample demo
* Remove the headset and let it sleep
* Activate the headset again and resume the launch (hide the dev window)

I tried to replicate it more times but it worked ok.

**Comment by bluemarvin** [comment]
I only see this issue on the Pixel. When I try the same code on the Lenovo Mirage immersive mode works as expected.

**Comment by bluemarvin** [comment]
I was able to fix the Pixel issue as follows:

```c++
void
ExternalBlitter::EndFrame() {
  /*
  if (m.surface) {
    // We need to detach the SurfaceTexture to prevent the Gecko WebGL compositor from getting blocked.
    m.surface->DetachFromGLContext();
  }
   */
  if (m.surface) {
    m.surface->ReleaseTexImage();
    m.surface = nullptr;
  }
}

void
ExternalBlitter::StopPresenting() {

  if (m.surface) {
    m.surface->ReleaseTexImage();
    m.surface = nullptr;
  }
  for (std::map<const int32_t, GeckoSurfaceTexturePtr>::iterator iter = m.surfaceMap.begin(); iter != m.surfaceMap.end(); iter++) {
    if (iter->second) {
      iter->second->DetachFromGLContext();
    }
  }
  m.surfaceMap.clear();
}
```

After talking with @snorp, it seems that from his experience, calling DetachFromGLContext() after each frame is not reliable.

**Comment by bluemarvin** [comment]
I will test this on other platforms tomorrow.

**Comment by bluemarvin** [comment]
This should be fixed by #346
